scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -242,6 +243,55 @@
 
   (restore-offers [this pool-name offers])
 
+  (autoscaling? [_]
+    (and
+      (some-> synthetic-pods-config :image count pos?)
+      (some-> synthetic-pods-config :user count pos?)
+      (some-> synthetic-pods-config :max-pods-outstanding pos?)))
+
+  (autoscale! [this pool-name task-requests]
+    (try
+      (assert (cc/autoscaling? this)
+              (str "In " name " compute cluster, request to autoscale despite invalid / missing config"))
+      (let [outstanding-synthetic-pods (->> @all-pods-atom
+                                            (all-pods this)
+                                            (filter controller/synthetic-pod-job-uuid))
+            num-synthetic-pods (count outstanding-synthetic-pods)
+            {:keys [image user command max-pods-outstanding] :or {command "exit 0"}} synthetic-pods-config]
+        (log/info "In" name "compute cluster, there are" num-synthetic-pods
+                  "outstanding synthetic pod(s), and a max of" max-pods-outstanding "are allowed")
+        (if (>= num-synthetic-pods max-pods-outstanding)
+          (log/info "In" name "compute cluster, cannot launch more synthetic pods")
```

Comment:
Should we take the top `max-pods-outstanding` from the `outstanding-synthetic-pods` pods to process in this scenario?

Reply:
This logs when the number of outstanding synthetic pods is already >= the max.

We are actually doing what you describe below.

```
(take (- max-pods-outstanding num-synthetic-pods))
```